SimCore/src/SimCore/Simulator.cxx

```diff
@@ -157,6 +158,15 @@ void Simulator::produce(framework::Event& event) {
 
   event_header.setStringParameter("eventSeed", stream.str());
 
+  auto event_info = static_cast<UserEventInformation*>(
+      runManager_->GetCurrentEvent()->GetUserInformation());
+
+  auto hepmc3_events = event_info->getHepMC3GenEvents();
+  for (auto& hepmc3ev : hepmc3_events) {
+    hepmc3ev.event_number = event.getEventHeader().getEventNumber();
+  }
+  event.add("SimHepMC3Events", hepmc3_events);
```

Comment:
I'm curious on if we could avoid including this collection when not running the GENIE generator. Right now, this looks like we will be carrying around an empty branch with all of our other run configurations and I think that is a waste of space and potentially confusing. Could this `event.add` call be skipped if `hepmc3_events` is empty?